Prepare for release (#54)
0xPolygonZero/zk_evm: README.md

# zk_evm

> A collection of libraries to prove Ethereum blocks with Polygon Zero Type 1 zkEVM,

Reviewer: I think we tend to write `Type-n` rather than `Type n`.

> powered by starky and plonky2 proving systems.

Reviewer: links to starky and plonky2?

## Directory structure

> This repository contains several Rust crates:

Reviewer: ```suggestion
This repository contains the following Rust crates:
```

* [mpt_trie](./mpt_trie/README.md): A collection of types and functions to work with Ethereum Merkle Patricie Tries.

> * [trace_decoder](./trace_decoder/README.md): Flexible protocol designed to process Ethereum clients trace payloads into an IR format that can be

Reviewer: I'd avoid "value ladden" terms like "flexible" here (if it's flexible or not kind of depends on who is using it and for what purpose). Also not sure if it's fair to call it a "protocol" so I'd avoid that term to. How about: "trace_decoder: transform Ethereum execution traces into the IR format that the zkEVM prover understands" (with a link to the IR spec).

Author: We may want to reword the individual READMEs then, I basically copied some portions here.

Reviewer 2: I don't disagree with the premise that spending time rewriting the READMEs is probably worthwhile at some point, I'm going to argue that it's not the most important thing right now. "Good enough" is probably a fair aim for the time being. We can create a note on the open issues that deal with documentation stuff to deal with the READMEs too at some point in the future.

Reviewer: I think READMEs are like the driveway to your house. Not critical to the function of the abode, but it is kind of nice to have welcoming first approach for guests coming to visit.

I also think that making improvements iteratively works best, and would have zero qualms editing this README here, and do the others another day.

No strong opinion on this, don't let this slow down the release. :)

understood by the zkEVM prover.

* [evm_arithmetization](./evm_arithmetization/README.md): Defines all the STARK constraints and recursive circuits to generate succinct proofs of EVM execution.
It uses starky and plonky2 as proving backend: https://github.com/0xPolygonZero/plonky2.

> * [proof_gen](./proof_gen/README.md): A convenience library for generating proofs from inputs in Intermediate Representation (IR) format.

Reviewer: ```suggestion
* [proof_gen](./proof_gen/README.md): A convenience library for generating proofs from inputs already in Intermediate Representation (IR) format.
```

## Documentation

> Although documentation is still at its early stage and being currently worked on, useful material can

Reviewer: ```suggestion
Documentation is still incomplete and will be improved over time, a lot of useful material can
```
